src/main/java/net/datafaker/transformations/YamlTransformer.java

```diff
@@ -0,0 +1,100 @@
+package net.datafaker.transformations;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.StringJoiner;
+
+import net.datafaker.sequence.FakeSequence;
+
+public class YamlTransformer<IN> implements Transformer<IN, CharSequence> {
+
+    private static final String INDENTATION = "  ";
+
+    @Override
+    public CharSequence apply(IN input, Schema<IN, ?> schema) {
+        Field<IN, ?>[] fields = schema.getFields();
+
+        if (fields.length == 0) {
+            return "";
+        }
+        StringBuilder sb = new StringBuilder();
+        return apply(sb, input, fields, "");
+    }
+
+    @Override
+    public String generate(FakeSequence<IN> input, Schema<IN, ?> schema) {
+        if (input.isInfinite()) {
+            throw new IllegalArgumentException("The sequence should be finite of size");
+        }
+
+        StringJoiner data = new StringJoiner(LINE_SEPARATOR);
+        for (IN in : input) {
+            data.add(apply(in, schema));
+        }
+
+        return data.toString();
+    }
+
+    @Override
+    public String generate(Schema<IN, ?> schema, int limit) {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < limit; i++) {
+            sb.append(apply(null, schema));
+            if (i < limit - 1) {
+                sb.append("\n");
```

Comment:
Probably here should be `LINE_SEPARATOR` as well
```suggestion
                sb.append(LINE_SEPARATOR);
```

Reply:
Replaced line separator